internal/services/storage/storage_account_resource.go

```diff
@@ -1367,8 +1383,23 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 			return fmt.Errorf("expanding `queue_properties`: %+v", err)
 		}
 
-		if err = queueClient.UpdateServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName, queueProperties); err != nil {
-			return fmt.Errorf("updating Queue Properties: %+v", err)
+		// wait for queue service endpoint to become available (issue when re-creating a storage account)
+		log.Printf("[DEBUG] waiting for %s queue service to become available", id.ID())
+		stateConf := &pluginsdk.StateChangeConf{
+			Pending: []string{"Pending"},
+			Target:  []string{"Available"},
+			Refresh: func() (interface{}, string, error) {
+				if err = queueClient.UpdateServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName, queueProperties); err != nil {
+					return handleStorageServiceError("updating", "queue", err)
+				}
+				return true, "Available", nil
+			},
+			MinTimeout: 10 * time.Second,
+			Timeout:    d.Timeout(pluginsdk.TimeoutCreate),
```

Comment:
rather than using a custom timeout, we need to use the remaining timeout on the context, to ensure that we're not staggering timeouts e.g.:

```suggestion
		deadline, ok := ctx.Deadline()
		if !ok {
			return fmt.Errorf("internal-error: context had no deadline")
		}
		stateConf := &pluginsdk.StateChangeConf{
			Pending: []string{"Pending"},
			Target:  []string{"Available"},
			Refresh: func() (interface{}, string, error) {
				if err = queueClient.UpdateServiceProperties(ctx, id.ResourceGroupName, id.StorageAccountName, queueProperties); err != nil {
					return handleStorageServiceError("updating", "queue", err)
				}
				return true, "Available", nil
			},
			MinTimeout: 10 * time.Second,
			Timeout:    time.Until(deadline),
```

Reply:
this has been fixed in all occurrences